addon/helpers/is-component-definition.js

```diff
@@ -12,9 +12,15 @@ export function isComponentDefinition(content) {
   if (symbolPropKeys?.length) {
     let isGlimmerComponentDefinition = symbolPropKeys.some((symbolPropKey) => {
       let propValue = content[symbolPropKey];
+      // The check for `__container__` is due to an update in Ember 3.25.1 where ComponentClass
+      // is no longer a key in the INNER symbol. As far as I can tell the combination of Symbols
+      // and having a `__container__` key is specific only to components. Passing other objects
+      // such as JS objects, Ember objects, hashes, or services, etc, do not share these traits.
       return (
         propValue &&
-        Object.keys(propValue).some((key) => key === 'ComponentClass')
+        Object.keys(propValue).some(
+          (key) => key === 'ComponentClass' || key === '__container__'
```

Comment:
@andrewpye tagged you to get your thoughts here since you recently updated this for Glimmer components. This update has nothing to do with the node upgrade here, but with upcoming Ember 3.25, the ember try `ember-release` fails because there's an update that removed the `ComponentClass` key from the list of keys where we originally pulled it from.

This is super hacky but I dug around for a while into the new Class that gets passed in as the `propValue` here and I can't really find anything specific to it being a component anymore 😞

Reply:
Dear god, this just keeps getting worse doesn't it? 😅 If this works then I guess we don't have much choice, _but_ I'm wondering if it's better done as a separate PR with a title that will make it more traceable in the future if we need to dig into how we got this file into such a mess (not a criticism of your code, just an observation that this file started out as a seemingly harmless hack but turned out to not be at all future-proof 😉)

Reply:
Yeahh 😭 Good call, I can make it a separate PR.